extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -635,13 +635,13 @@ export class ProjectsController {
 	private async promptForNewObjectName(itemType: templates.ProjectScriptType, _project: Project, folderPath: string, fileExtension?: string): Promise<string | undefined> {
 		const suggestedName = itemType.friendlyName.replace(/\s+/g, '');
 		let counter: number = 0;
-		let pathExists: boolean = true;
+		let pathExists;
 
-		while (counter < Number.MAX_SAFE_INTEGER && pathExists) {
+		do {
 			counter++;
 			const extension: string = fileExtension ? fileExtension : '';
```

Comment:
Nice little bit of syntactic sugar you can use here:

`const extension = fileExtension ?? '';`

Called a "null coalescing operator", it uses the left-side value if it's defined, otherwise it uses the right-side value.  One of my favorite bits.

Reply:
A couple ideas for ways to streamline this little chunk of code:

* move the `const extension` declaration outside of the do-while loop since it doesn't need to be recomputed each time
* do away with it entirely since the null coalescing operator is nice and concise: `pathExists = await utils.exists(path.join(folderPath, suggestedName + counter + (fileExtension ?? '')));`
* do away with _that_ entirely by moving it into the while loop's conditional:

```
do {
    counter++;
} while (counter < Number.MAX_SAFE_INTEGER
    && await utils.exists(path.join(folderPath, suggestedName + counter + (fileExtension ?? ''))));
```

All depends on whatever balance between readability and concise code you think is best!  I like 2 or 3, personally.

Reply:
thank you @Benjin ! this syntactic sugar is sweet :)